### PR TITLE
[Docs] A Design Doc for the Public Proposal Process

### LINKED ISSUE #4888: Formalize and document a process for Proposals at Magma
Now that Magma is transitioning to the Linux Foundation, it is increasingly important that we offer a clear public process and venue for Proposals, their discussion and outcome handling. Proposals can cover all sorts of aspects of Magma - including system design, feature or API design, processes for code submittal, repository changes, and more.

The process:

- Needs to be clearly documented
- Needs to leave a trace of historic Proposal outcomes (both accepted and rejected)
- Needs to be fair and offer equal opportunity for all parties to have their ideas heard
- Needs to have Magma points-of-contact who shepherd all Proposals through review
- Needs to be lightweight and timely in execution

Having surveyed the landscape of this problem space (Open source processes for Proposal handling) - I here propose that we adopt a variant of the [Golang community Proposal process](https://github.com/golang/proposal).

### PR DESCRIPTION
**Pull request by electronjoe**
A Design Document for the Proposal outlined in Issue #4888.

Following the process outlined in this very document, here we submit for review a proposal for a Public Proposal Process.

Comments on this PR should be limited to grammar, spelling, or procedural errors related to the preparation of the proposal itself. Commit of proposal design documents should be low-friction - their review occurs in the associated GitHub Issues.

### CODE REVIEW

File: docs/readmes/proposals/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+Proposals may take all sorts of shapes, including:
+
+- Design Proposals
+- Repository change Proposals
+- Process Proposals
+
+## Magma Proposal Process
+
+Magma has elected to base its Proposal Process of the
```
Comment (andreilee) [inline docs/readmes/proposals/README.md]:
```suggestion
Magma has elected to base its Proposal Process off the
```

File: docs/readmes/proposals/README.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+id: proposal_process
+title: Magma Proposals
+hide_title: true
+---
+# Magma Proposals
+
+As an open source project, it is important that all contributors are able to
+observe substantitive Proposals, comment on them, and be able to discover
+historical discussions.
+This preserves important hisotrical context for decisions, reduces load on
```
Comment (ardzoht) [inline docs/readmes/proposals/README.md]:
```suggestion
This preserves important historical context for decisions, reduces load on
```

File: docs/readmes/proposals/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+The principal goal of the review meeting is to make sure that proposals
+are receiving attention from the right people, by cc'ing relevant developers,
+raising important questions, pinging lapsed discussions, and generally trying
+to guide discussion toward agreement about the outcome.
+The discussion itself is expected to happen on the issue tracker,
+so that anyone can take part.
+
+A group of Magma team members holds “proposal review meetings”
+approximately weekly to review pending proposals.
```
Comment (ardzoht) [inline docs/readmes/proposals/README.md]:
```suggestion
approximately on a weekly basis to review pending proposals.
```

**Comment by electronjoe** [comment]
PTAL, I would like to submit this Design Document draft and feel it has been scrubbed appropriately for typos.

**Review by ardzoht — APPROVED** [review]

**Pull request merged**